File: Chapter 04/C#/Section_4_3_Cross_Cutting_Concerns/Before/OrderService.cs
namespace Chapter04.Section_4_3.Before
{
    public class OrderService
    {
        public void SaveOrder(Order order)
        {
            // This is a hidden, rigid dependency.
            StaticFileLogger.Log("Saving order...");
            Console.WriteLine("(BEFORE_SERVICE) Order saved.");
        }
    }
}

File: Chapter 04/Python/section_4_3_cross_cutting_concerns/before/order_service.py
from .static_file_logger import StaticFileLogger

class OrderService:
    """
    ARCHITECTURE WARNING: This class has a hidden dependency.
    It relies on an external global logger, which prevents 
    it from being a "Pure" business logic component.
    """
    def save_order(self, order):
        # 🚨 VIOLATION: Hidden, rigid dependency.
        StaticFileLogger.log("Saving order...")
        print("(BEFORE_SERVICE) Order saved.")